File: task-03/Go/subtask-04.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"strconv"
	"strings"
)

func createStarPattern(n int) string {
	pattern := ""

	for i := 1; i <= n; i += 2 {
		padding := strings.Repeat(" ", (n-i)/2)
		stars := strings.Repeat("*", i)
		pattern += padding + stars + "\n"
	}

	for i := n - 2; i >= 1; i -= 2 {
		padding := strings.Repeat(" ", (n-i)/2)
		stars := strings.Repeat("*", i)
		pattern += padding + stars + "\n"
	}

	return pattern
}

func main() {
	content, err := ioutil.ReadFile("input2.txt")
	if err != nil {
		log.Fatal(err)
	}

	n, err := strconv.Atoi(strings.TrimSpace(string(content)))
	if err != nil {
		log.Fatal(err)
	}

	pattern := createStarPattern(n)
	err = ioutil.WriteFile("output2.txt", []byte(pattern), 0644)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("Pattern written to output2.txt")
}

File: task-03/Haskell/subtask-04.hs
import System.IO

createStarPattern :: Int -> String
createStarPattern n = 
    let upper = concatMap (\i -> replicate ((n - i) `div` 2) ' ' ++ replicate i '*' ++ "\n") [1,3..n]
        lower = concatMap (\i -> replicate ((n - i) `div` 2) ' ' ++ replicate i '*' ++ "\n") [n-2,n-4..1]
    in upper ++ lower

main :: IO ()
main = do
    content <- readFile "input2.txt"
    let n = read content :: Int

    let pattern = createStarPattern n
    writeFile "output2.txt" pattern
